pickup/lib/pickupBatch.js

```
@@ -0,0 +1,106 @@
+      await upload({ body, cid, downloadError })
+      logger.info({ cid, requestid, messageId: message.MessageId }, 'Car downloaded and stored in S3')
+
+      // After the download some low level action are required to override the `sqs-consumer` library limit.
+      // The library works with the full batch and the is required to remove the messages while they are processed.
+      // The processed message is removed from the queue list to avoid further changeVisabilityTimeout.
+      const arrayRemoveIndex = requestIds.indexOf(requestid)
+      requestIds.splice(Number(arrayRemoveIndex), 1)
+      messages.splice(Number(arrayRemoveIndex), 1)
+      logger.trace({ cid, requestid, messageId: message.MessageId, arrayRemoveIndex }, 'Removed processed message from the messages')
+
+      await queueManager.deleteMessage(message)
+
+      resultStats[cid] = 'success'
+    } catch (err) {
+      // The processed message is removed from the queue list to avoid further changeVisabilityTimeout.
+      const arrayRemoveIndex = requestIds.indexOf(requestid)
+      requestIds.splice(Number(arrayRemoveIndex), 1)
+      messages.splice(Number(arrayRemoveIndex), 1)
+
+      if (downloadError.code === ERROR_TIMEOUT) {
+        logger.info({ cid, requestid, messageId: message.MessageI, arrayRemoveIndex }, 'Download timeout')
+        // Update the status on dynamodb to failed
+        await updatePinStatus(dynamo, dynamoTable, cid, 'failed')
+
+        // Delete the message from the queue
+        await queueManager.deleteMessage(message)
+        resultStats[cid] = 'timeout'
+      } else {
+        // For any other error the message from the queue is not removed,
+        // then when the visibility timeout is expired the file is resend on the queue
+        // A deadLetterQueue should be set to avoid infinite retry.
+        logger.error({ err, cid, requestid, messageId: message.MessageI, arrayRemoveIndex }, 'Download error')
```

the original error is swallowed, we could either add `logger.error` at `fetchCar` throw or handle `cause` in the logger err serializiation